pyClient/zeth/joinsplit.py

```diff
@@ -0,0 +1,723 @@
+from __future__ import annotations
+import zeth.constants as constants
+import zeth.errors as errors
+from zeth.utils import get_trusted_setup_dir, hex_extend_32bytes, \
+    hex_digest_to_binary_string, string_list_flatten, encode_single, \
+    encode_abi, encrypt, decrypt, get_public_key_from_bytes
+from zeth.prover_client import ProverClient
+from api.util_pb2 import ZethNote, JoinsplitInput, HexPointBaseGroup1Affine, \
+    HexPointBaseGroup2Affine
+from nacl.public import PrivateKey, PublicKey  # type: ignore
+import nacl.utils  # type: ignore
+import api.prover_pb2 as prover_pb2
+
+import os
+import json
+import sys
+from Crypto import Random
+from hashlib import blake2s, sha256
+from py_ecc import bn128 as ec
+from typing import Tuple, Dict, List, Iterable, Union, Any
+
+
+FQ = ec.FQ
+G1 = Tuple[ec.FQ, ec.FQ]
+
+
+class ApkAskPair:
+    def __init__(self, a_sk: str, a_pk: str):
+        self.a_pk = a_pk
+        self.a_sk = a_sk
+
+
+SK = Tuple[int, int]
```

Comment:
Why not, `SK = Tuple[int, int]` -> `SK = Tuple[FQ, FQ]`?

Reply:
Changed